tasks/angular/forms.md

```diff
@@ -41,38 +41,51 @@ The main goal of this task is to implement forms handling and validation using R
-            - check for the URL validity (message: "The image link is invalid")
         - link to the video
             - required (message: "Please enter a link to the video")
-            - check for the URL validity (message: "The video link is invalid")
         - creation date
             - required (message: "Please enter a creation date")
             - an entered date should not be in the future (message: "The date is invalid")
-
-- The application should prevent a user from submitting a form in case one of the described rules is failed
-- In case a user has entered invalid data into an input, the border should be colored in red
+    - implement "Tags" sub-form:
+        - initially a single "Tag" input is present
+        - click "Add tag" button to add an extra input
+        - each tag input should be required
+        - form can have up to five "Tag" inputs
+    - implement "Reset" button
+        - when clicked all inputs should be reset to their initial values and the "Tags" sub-form should return to having only a single (empty) input
+
+- The application should prevent user from submitting the form if at least one of the described validation rules didn't pass
+- If the user has entered invalid data into an input, its border should be colored in red
 - To indicate which rule fails, an appropriate message under an invalid input field should be displayed.
 
 #### Evaluation criteria
-Maximum points - **100**
+Maximum score - **100**
 
 - [ ] **Login block** is implemented as reactive form (**+10**)
-- [ ] Validation rules are applied to the **Login block**. The form submission is possible only if a user enters valid data to all inputs (**+20**)
-- [ ] The application indicated which inputs are invalid on the **Login block** by applying appropriate style and showing corresponding message (**+20**)
+- [ ] Validation rules are applied to the **Login block**. Form submission is possible only if all inputs are valid (**+15**)
+- [ ] The application indicates which inputs are invalid on the **Login block** with appropriate styles and messages (**+15**)
 - [ ] **Admin page** is implemented as reactive form (**+10**)
-- [ ] Validation rules are applied to the **Admin page**. The form submission is possible only if a user enters valid data to all inputs (**+20**)
-- [ ] The application indicated which inputs are invalid on the **Admin page** by applying appropriate style and showing corresponding message (**+20**)
+- [ ] Validation rules are applied to the **Admin page**. Form submission is possible only if all inputs are valid (**+15**)
+- [ ] The application indicates which inputs are invalid on the **Admin page** with appropriate styles and messages (**+15**)
+- [ ] "Tags" sub-form is implemented using `FormArray`. "Add tag" button works correctly (**15**)
+- [ ] "Reset" button functionality is fully implemented (**5**)
+- [ ] TODO: seems that form submit logic is missing at any point of the course. To be added if confirmed.
```

Comment:
Login form and Card creation form (https://github.com/rolling-scopes-school/tasks/blob/master/tasks/angular/NgRX.md) is used in application.
Item have to be removed

Reply:
Did it.